tests/x11/vino.pm

```diff
@@ -0,0 +1,66 @@
+# Copyright (C) 2018 SUSE LLC
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, see <http://www.gnu.org/licenses/>.
+#
+#
+#
+# Summary: Test if the Screen Sharing GNOME Desktop functionality works
+#   We currently need to install vino package for that
+# Maintainer: Pavel Dostál <pdostal@suse.cz>
+
+use base "x11test";
+use strict;
+use testapi;
+use utils;
+
+sub run {
+    record_info 'vino missing',     'After the installation the screen sharing is not available - vino is missing.';
+    assert_and_click 'upper_right', 'left', 90;
+    assert_and_click 'settings',    'left';
+    assert_and_click 'sharing',     'left';
+    assert_screen 'without_screensharing';
+    send_key 'ctrl-q';
+
+    record_info 'Install vino', 'We are going to install vino.';
+    mouse_hide;
+    ensure_installed 'vino';
+
+    record_info 'Relogin',          'We must log off and back in to see the change.';
+    assert_and_click 'upper_right', 'left';
+    assert_and_click 'user',        'left';
+    assert_and_click 'log_out';
+    assert_and_click 'really-log-out';
+    assert_and_click 'click-username';
+    type_string 'nots3cr3t';
+    send_key 'ret';
+
+    record_info 'vino present',     'Now we gonna check that the screen sharing is available';
+    assert_and_click 'upper_right', 'left';
+    assert_and_click 'settings',    'left';
+    assert_and_click 'sharing',     'left';
+    assert_screen 'with_screensharing';
+    send_key 'ctrl-q';
+}
+
+sub test_flags {
+    # 'fatal'          - abort whole test suite if this fails (and set overall state 'failed')
+    # 'ignore_failure' - if this module fails, it will not affect the overall result at all
+    # 'milestone'      - after this test succeeds, update 'lastgood'
+    # 'norollback'     - don't roll back to 'lastgood' snapshot if this fails
+    return {fatal => 1};
```

Comment:
I think you misunderstand the meaning of "fatal". I suggest you just delete the whole sub "test_flags" as the default are most likely what you want. The module can still "fail" and would subsequently mark the whole job as failed as well if this module (at least) fails. "fatal" would mean there is no sense to continue testing other - most likely unrelated - test modules which I doubt is what you want. Consider the installation steps, they are "fatal" because if the installation does not finish there is no point trying to start a "firefox" test. But if the firefox test module fails one can of course still try to run the "chromium" test. But if firefox fails and if the test continues to conduct other test modules the "overall state" would still be "failed" even though firefox is not marked as fatal.